qiita_db/search.py

```diff
@@ -303,9 +354,9 @@ def _parse_study_search_string(self, searchstr,
                     meta_header_type_lookup[header] = 'varchar'
 
         # create the study finding SQL
-        # remove metadata headers that are in required_sample_info table
-        meta_headers = meta_headers.difference(self.required_cols).difference(
-            self.study_cols)
+        # remove metadata headers that are in non-dynamic tables
+        meta_headers = meta_headers.difference(
```

Comment:
should `self.required_samp_cols` be a `list`? It likely should be at least immutable. Less sure about `self.study_cols` but I'd assume the same